### PR TITLE
Test listProj in semtype testing

### PR DESCRIPTION
**Pull request by ushirask**
## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

**Review by jclark — CHANGES_REQUESTED** [review]
tests are failing in typeDescToWords

File: compiler/testSuite/semtype/proj2-t .bal [hunk 1]
```diff
@@ -0,0 +1,5 @@
+type I int;
+
+// @type A[0] = I
+// @type A[1] = I
```
Comment (jclark) [inline compiler/testSuite/semtype/proj2-t .bal]: We want to test `A[int]` too.

File: compiler/tests/testSuite.bal [hunk 2]
```diff
@@ -212,8 +212,8 @@ function resolveTestSemtype(t:Context tc, map<t:SemType> m, s:Identifier|s:TypeP
         int|s:Identifier index = tn.index;
         if t:isSubtypeSimple(t, t:LIST) {
             if index is int {
-                return t:listMemberType(tc, t, index);
-                //return t:listProj(tc, t, index);
+                //return t:listMemberType(tc, t, index);
+                return t:listProj(tc, t, index);
```
Comment (jclark) [inline compiler/tests/testSuite.bal]: We need to test listMemberType too. I suggest:

- call both (in cases where index is a singleton)
- check that result from listProj is right
- check that result from listMemberType is a supertype of result from listProj

**Review by jclark — CHANGES_REQUESTED** [review]

File: compiler/testSuite/semtype/proj2-t .bal (hunk 1 above)
Comment (ushirask) [reply compiler/testSuite/semtype/proj2-t .bal]: can we test this with `listProj` since we need  the key value for listProj?

Comment (jclark) [reply compiler/testSuite/semtype/proj2-t .bal]: Oh yes, you're right. We can't do that.

Comment (jclark) [reply compiler/testSuite/semtype/proj2-t .bal]: We can only test that with listMemberType which will give us `int?`

Comment (ushirask) [reply compiler/testSuite/semtype/proj2-t .bal]: We could add 
```
type I int;
type I2 int?;

// @type A[I] = I2
```
just for testing purposes

Comment (jclark) [reply compiler/testSuite/semtype/proj2-t .bal]: Yes.

**Comment by jclark** [comment]
Is this ready for review?

**Comment by ushirask** [comment]
> Is this ready for review?

yes

**Review by jclark — APPROVED** [review]

**Pull request merged**